TestSwiftyDropbox/TestSwiftyDropbox_iOS/FileBrowserView.swift

```diff
@@ -4,66 +4,84 @@
 
 import SwiftUI
 
+@available(iOS 16.0, *)
+func MakeFileBrowserView(localURL: URL?) -> FileBrowserView {
+    FileBrowserView(viewModel: FileBrowserViewModel(localURL: localURL))
+}
+
+@available(iOS 16.0, *)
+class FileBrowserViewModel: ObservableObject {
+    let localURL: URL?
+    @Published var files: [URL] = []
+    @Published var showAlert = false
+    @Published var selectedFileContent: String = ""
+
+    init(localURL: URL?) {
+        self.localURL = localURL
+
+        let fileManager = FileManager.default
+
+        guard let localURL = localURL else {
```

Comment:
Should this be a failable initializer instead?

Reply:
In this case we're trying to make a view model, just one that doesn't actually have file access